public/app/features/trails/DataTrail.tsx

```diff
@@ -208,12 +287,321 @@ export class DataTrail extends SceneObjectBase<DataTrailState> {
     this.setState(stateUpdate);
   }
 
+  /**
+   * Check that the data source has otel resources
+   * Check that the data source is standard for OTEL
+   * Show a warning if not
+   * Update the following variables:
+   * deployment_environment (first filter), otelResources (filters), otelJoinQuery (used in the query)
+   * Enable the otel experience
+   *
+   * @returns
+   */
+  public async checkDataSourceForOTelResources() {
+    // call up in to the parent trail
+    const trail = getTrailFor(this);
+
+    // get the time range
+    const timeRange: RawTimeRange | undefined = trail.state.$timeRange?.state;
+
+    if (timeRange) {
+      const otelResourcesVariable = sceneGraph.lookupVariable(VAR_OTEL_RESOURCES, this);
+      const otelDepEnvVariable = sceneGraph.lookupVariable(VAR_OTEL_DEPLOYMENT_ENV, this);
+      const otelJoinQueryVariable = sceneGraph.lookupVariable(VAR_OTEL_JOIN_QUERY, this);
+      const filtersVariable = sceneGraph.lookupVariable(VAR_FILTERS, this);
+
+      const datasourceUid = sceneGraph.interpolate(trail, VAR_DATASOURCE_EXPR);
+
+      const otelTargets = await totalOtelResources(datasourceUid, timeRange);
+      const deploymentEnvironments = await getDeploymentEnvironments(datasourceUid, timeRange);
+      const hasOtelResources = otelTargets.job !== '' && otelTargets.instance !== '';
+      // const isStandard = await isOtelStandardization(datasourceUid, timeRange);
+      if (
+        otelResourcesVariable instanceof AdHocFiltersVariable &&
+        otelDepEnvVariable instanceof CustomVariable &&
+        otelJoinQueryVariable instanceof ConstantVariable &&
+        filtersVariable instanceof AdHocFiltersVariable
+      ) {
+        // HERE WE START THE OTEL EXPERIENCE ENGINE
+        // 1. Set deployment variable values
+        // 2. update all other variables and state
+        // Note** isStandard is not reliable because of an issue where there are
+        // multiple target_info series with job&instance pairs due to staleness which breaks the join.
+        // We handle this by now using topk to choose 1 series to allow for the join
+        if (hasOtelResources && deploymentEnvironments.length > 0 /*&& isStandard*/) {
```

Comment:
`isStandard`, do we need this?

Reply:
Deleting it! This is helpful knowledge but we do not need it in the code.